src/rgw/rgw_sal_motr.cc

```diff
@@ -2739,10 +2753,19 @@ int MotrMultipartUpload::init(const DoutPrefixProvider *dpp, optional_yield y,
   if (rc < 0)
     return rc;
   string tenant_bkt_name = get_bucket_name(bucket->get_tenant(), bucket->get_name());
+  string upload_id = get_upload_id();
```

Comment:
"upload_id" will never be empty here (if we are able to generate it successfully during multipart init call). So mostly, the code from line# 2757 below will never execute (dead code). You may remove the "if" block from line# 2758.

Also this is multipart operation call, and store->get_upload_id() will try to get some data from bucket index. During multipart, we read from multipart index, and not from bucket index.

Reply:
Done.